geoportailv3/static/js/layerinfo/layerinfoservice.js

```diff
@@ -14,32 +14,112 @@ goog.require('ngeo.CreatePopup');
 app.ShowLayerinfo;
 
 
+
 /**
  * @param {angular.$http} $http Angular $http service
  * @param {angular.$sce} $sce Angular $sce service
+ * @param {angular.Scope} $rootScope The rootScope provider.
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
  * @param {ngeo.CreatePopup} ngeoCreatePopup Ngeo popup factory service
- * @return {app.ShowLayerinfo} The show layer info function.
+ * @constructor
  * @ngInject
  */
-app.showLayerinfoFactory = function($http, $sce, ngeoCreatePopup) {
+app.showLayerinfoFactory = function($http, $sce, $rootScope,
+    gettextCatalog, ngeoCreatePopup) {
 
   /**
    * @type {ngeo.Popup}
    */
   var popup = ngeoCreatePopup();
+  /**
+   * @type {Object.<string, !angular.$q.Promise>}
+   * @private
+   */
+  var promises_ = {};
+
+  /**
+   * @type {ol.layer.Layer}
+   */
+  var currentLayer = null;
+
+  $rootScope.$on('gettextLanguageChanged', function() {
+    if (!goog.isNull(currentLayer) && popup.getOpen()) {
+      showLayerInfo(currentLayer);
+    }
+  });
+
+  return showLayerInfo;
+
+  /**
+   * @param {ol.layer.Layer} layer The layer
+   * @this {Object}
+   */
+  function showLayerInfo(layer) {
+    currentLayer = layer;
+    var title = /** @type {string} */ (layer.get('label'));
+    var localMetadata = /** @type {Object.<string>} */
+        (layer.get('metadata'));
+    var metadataUid = /** @type {string} */
+        (localMetadata['metadata_id']);
+    popup.setTitle(title);
+    var currentLanguage = /** @type {string} */
+        (gettextCatalog.currentLanguage);
```

Comment:
[`gettextCatalog.currentLanguage is a `string`](https://github.com/camptocamp/ngeo/blob/master/externs/angular-gettext.js#L32) so you don't need a type-cast for the local `currentLanguage` variable. Either you don't provide a type at all, or you use that syntax:

```js
  /** @type {string} */
  var currentLanguage = gettextCatalog.currentLanguage;
```

Another possible syntax is:

```js
  var /** @type {string} */ currentLanguage = gettextCatalog.currentLanguage;
```

But probably providing a type is not necessary here.